app/workers/work-reaper.ts

```diff
@@ -25,48 +25,99 @@ export default class WorkReaper implements Worker {
     this.logger = config.logger;
   }
 
-  async deleteTerminalWork(notUpdatedForMinutes: number, jobStatus: JobStatus[]): Promise<void> {
-    try {
-      const workItemIds = await getWorkItemIdsByJobUpdateAgeAndStatus(
-        db, notUpdatedForMinutes, jobStatus,
-      );
-      if (workItemIds.length) {
-        const chunkedWorkItemIds = _.chunk(workItemIds, batchSize);
-        for (const workItemIdsChunk of chunkedWorkItemIds) {
-          const numItemsDeleted = await deleteWorkItemsById(db, workItemIdsChunk);
-          this.logger.info(`Work reaper removed ${numItemsDeleted} work items`);
+
+  /**
+   * Find work items that are older than notUpdatedForMinutes and delete them.
+   * @param notUpdatedForMinutes - upper limit on the duration since the last update
+   * @param jobStatus - a list of terminal job statuses
+   * @returns Resolves when the request is complete
+   */
+  async deleteTerminalWorkItems(notUpdatedForMinutes: number, jobStatus: JobStatus[]): Promise<void> {
+    let done = false;
+    let startingId = 0;
+    let totalDeleted = 0;
+    const batchSize = env.workReaperBatchSize;
+    this.logger.info('Work reaper delete terminal work items started.');
+
+    while (!done) {
```

Comment:
I think it would be a bit cleaner to use something like
```
try {
  let workItemIds;
  while ((workItemIds = await getWorkItemIdsByJobUpdateAgeAndStatus(
          db, notUpdatedForMinutes, jobStatus, startingId, batchSize,
        )).length > 0) {
...
 }
} catch(...)
```

Reply:
I am using `done` to not do an extra query on the last batch when the number of retrieved records is less than the batch size. I find putting too much in the `while` condition not as readable as the `done` flag.